src/diffusers/pipeline_utils.py

```diff
@@ -285,6 +287,21 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         # 1. Download the checkpoints and configs
         # use snapshot download here to get it working from from_pretrained
         if not os.path.isdir(pretrained_model_name_or_path):
+            config_dict = cls.get_config_dict(
+                pretrained_model_name_or_path,
+                cache_dir=cache_dir,
+                resume_download=resume_download,
+                proxies=proxies,
+                local_files_only=local_files_only,
+                use_auth_token=use_auth_token,
+                revision=revision,
+            )
+            # make sure we only download sub-folders and `diffusers` filenames
+            folder_names = [k for k in config_dict.keys() if not k.startswith("_")]
+            allow_patterns = [os.path.join(k, "*") for k in folder_names]
+            allow_patterns += [WEIGHTS_NAME, SCHEDULER_CONFIG_NAME, CONFIG_NAME, cls.config_name]
```

Comment:
Good catch! This would probably break ONNX currently no @anton-l ? Regarding Flax, I think we have to / should do a `FlaxDiffusionPipeline` anyways

Reply:
Actually, ONNX would always be saved in a folder, or have ONNX_WEIGHTS_NAME so we should be fine here once having added ONNX.

I would advocate to create a new `FlaxDiffusionPipeline` because functionality like `.to(...)` would otherwise only be useful for PyTorch and there is likely to come more such functionality. Similarly we would have a `jit` function on the `FlaxDiffusionPipeline` which is not useful for PyTorch, so I think it's better to create a new class here.

What do you think @patil-suraj ?

Reply:
Agree with Patrick, we should really add `FlaxDiffusionPipeline` for flax.

Reply:
I agree. But that pipeline would still be downloadable using this same function, wouldn't it? So we need to use the right filename here.

Reply:
I think we'd create a new function here as well :-)